### PR TITLE
Classification models implemented

### PR DESCRIPTION
**Pull request by will-jl944**
1. Add ResNet series, MobileNet series, AlexNet, DarkNet53. Already aligned with PaddleClas (CPU, single GPU, multi-GPUs, performance, time consumption).
2. Support user-defined optimizer.
3. Fix typo in UNet tutorial.

### CODE REVIEW

File: paddlex/cv/models/utils/seg_metrics.py [hunk 1]
```diff
@@ -16,6 +16,7 @@
 from scipy.sparse import csr_matrix
 import paddle
 import paddle.nn.functional as F
+from scipy.sparse import csr_matrix
```
Comment (FlyingQianMM) [inline paddlex/cv/models/utils/seg_metrics.py]: 上面有一句了

File: paddlex/cv/models/utils/seg_metrics.py [hunk 2]
```diff
@@ -163,3 +164,19 @@ def f1_score(intersect_area, pred_area, label_area):
             f1_sco = 2 * prec * rec / (prec + rec)
         class_f1_sco.append(f1_sco)
     return np.array(class_f1_sco)
+
+
```
Comment (FlyingQianMM) [inline paddlex/cv/models/utils/seg_metrics.py]: confusion_matrix有没有seg的paddle  op实现？

File: paddlex/utils/checkpoint.py [hunk 3]
```diff
@@ -34,7 +34,26 @@
...
+    'https://paddle-imagenet-models-name.bj.bcebos.com/dygraph/ResNet50_vd_pretrained.pdparams',
+    'ResNet101_vd_IMAGENET':
+    'https://paddle-imagenet-models-name.bj.bcebos.com/dygraph/ResNet101_vd_pretrained.pdparams',
+    'ResNet152_vd_IMAGENET':
+    'https://paddle-imagenet-models-name.bj.bcebos.com/dygraph/ResNet152_vd_pretrained.pdparams',
+    'ResNet200_vd_IMAGENET':
+    'https://paddle-imagenet-models-name.bj.bcebos.com/dygraph/ResNet200_vd_pretrained.pdparams',
+    'AlexNet_IMAGENET':
+    'https://paddle-imagenet-models-name.bj.bcebos.com/dygraph/AlexNet_pretrained.pdparams'
+}
```
Comment (FlyingQianMM) [inline paddlex/utils/checkpoint.py]: 预训练模型链接是不是没写全？比如darknet,mobilenet

File: paddlex/utils/checkpoint.py [hunk 4]
```diff
@@ -15,7 +15,7 @@
 import os
 import os.path as osp
 import paddle
-import paddlex
+from paddle.static import load_program_state
 import paddlex.utils.logging as logging
```
Comment (FlyingQianMM) [inline paddlex/utils/checkpoint.py:18]: load_program_state还没用到？delete

File: tutorials/train/image_classification/AlexNet.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+num_classes = len(train_dataset.labels)
+model = pdx.cls.AlexNet(num_classes=num_classes)
+
+model.train(
+    num_epochs=10,
+    train_dataset=train_dataset,
+    train_batch_size=32,
+    eval_dataset=eval_dataset,
+    pretrained_weights=None,
```
Comment (FlyingQianMM) [inline tutorials/train/image_classification/AlexNet.py]: pretrained_weights设置为None的原因是什么？
下面几个例子也是

File: paddlex/utils/checkpoint.py (hunk 3 above)
Comment (will-jl944) [reply paddlex/utils/checkpoint.py]: TODO

File: paddlex/cv/models/utils/seg_metrics.py (hunk 1 above)
Comment (will-jl944) [reply paddlex/cv/models/utils/seg_metrics.py]: Done

File: tutorials/train/image_classification/AlexNet.py (hunk 5 above)
Comment (will-jl944) [reply tutorials/train/image_classification/AlexNet.py]: TODO

File: paddlex/utils/checkpoint.py (hunk 4 above)
Comment (will-jl944) [reply paddlex/utils/checkpoint.py:18]: Done

File: paddlex/utils/checkpoint.py (hunk 3 above)
Comment (will-jl944) [reply paddlex/utils/checkpoint.py]: Done

File: tutorials/train/image_classification/AlexNet.py (hunk 5 above)
Comment (will-jl944) [reply tutorials/train/image_classification/AlexNet.py]: Done

File: paddlex/cls.py [hunk 6]
```diff
@@ -64,3 +64,10 @@
 Xception41 = cv.models.Xception41
 Xception65 = cv.models.Xception65
 Xception71 = cv.models.Xception71
+ShuffleNetV2_x0_25 = cv.models.ShuffleNetV2_x0_25
+ShuffleNetV2_x0_33 = cv.models.ShuffleNetV2_x0_33
+ShuffleNetV2_x0_5 = cv.models.ShuffleNetV2_x0_5
+ShuffleNetV2_x1_0 = cv.models.ShuffleNetV2_x1_0
+ShuffleNetV2_x1_5 = cv.models.ShuffleNetV2_x1_5
+ShuffleNetV2_x2_0 = cv.models.ShuffleNetV2_x2_0
+ShuffleNetV2_swish = cv.models.ShuffleNetV2_swish
```
Comment (jiangjiajun) [inline paddlex/cls.py:60]: ShuffleNet这里的x0_25不能类似MobileNetV2一样，作为Scale系数吗

Comment (will-jl944) [reply paddlex/cls.py:60]: 目前的实现是，scale是网络结构（class ShuffleNetV2）的一个参数，这里的暴露给用户的cv.models.xxx已经是model了，他们的父类是class BaseClassifier，BaseClassifier的__init__目前只接受model_name和num_classes两个参数，没有scale。
是否要把scale加进BaseClassifier，提供给用户？
